app/jobs/notify_job.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class NotifyJob < ApplicationJob
+  queue_as :webhooks
+
+  def perform(notification_id:)
+    notification = Notification.kept.find(notification_id)
+
+    data = ActiveModelSerializers::Adapter.create(NotificationSerializer.new(notification)).to_json
+    hmac = Encryptor.hmac(notification.subscription.secret, data)
+
+    response = client.post(notification.subscription.callback_url, data, 'PECS-SIGNATURE': hmac, 'PECS-NOTIFICATION-ID': notification_id)
```

Comment:
I feel like this may have been discussed already, so apologies. In case the subscription is disabled between the notification being queued and this job running, should we guard this post on `subscription.enabled?`

Reply:
@spikeheap very pertinent question - @ngw and I wanted to get a view from the team on this. To be clear, at the guard is in place here thanks to: `notification = Notification.kept.find(notification_id)` - the `.kept` filters out notifications which have a disabled subscription

Reply:
we should yes, commit incoming.

Reply:
todo: add subscription check

Reply:
that felt extremely weird actually :D